src/In.ProjectEKA.HipService/Link/LinkController.cs

```diff
@@ -46,11 +48,32 @@ public AcceptedResult LinkFor(LinkReferenceRequest request)
         public AcceptedResult LinkPatientFor(LinkPatientRequest request)
         {
             backgroundJob.Enqueue(() => LinkPatientCareContextFor(request));
+            //TODO: Remove from UnlinkedAccounts - Zafar
             return Accepted();
         }
 
         public async Task LinkPatient(LinkReferenceRequest request)
         {
+            var discoveryRequest = await discoveryRequestRepository.GetRequestFor(request.TransactionId);
+            request.Patient.ReferenceNumber = discoveryRequest.PatientReferenceNumber;
+            var (accounts, e) = await linkPatient.GetUnlinkedAccounts(discoveryRequest.PatientReferenceNumber);
+
+            if (e != null)
+            {
+                Log.Error(e);
+                //TODO: Handle exception - Zafar
```

Comment:
Both are not pairing?

Reply:
We are :). One name was enough to track.